### PR TITLE
convert  legacy token balances

### PR DESCRIPTION
**Pull request by szerintedmi**
show legacy token balances and option to convert them

### CODE REVIEW

**Comment by szerintedmi** [comment]
Deploy preview for *augmint* processing.

Built with commit f7dad53e485a3475653ca9e769c09453ae5f9e6c

https://app.netlify.com/sites/augmint/deploys/5ac207ad0b79b7349a3a3e1e

File: src/modules/legacyBalancesProvider.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    if (account.toLowerCase() === userAccount.toLowerCase()) {
+        console.debug(
+            "legacyBalancesProvider - LegacyTokenConverted event received for current user account. Dispatching refreshLegacyBalances()"
+        );
+        store.dispatch(refreshLegacyBalances());
+    }
+};
+
+const onUserAccountChange = (newVal, oldVal, objectPath) => {
+    if (newVal !== "?") {
```
Comment (reyrodrigez) [inline src/modules/legacyBalancesProvider.js]: is this condition for the initial account change?
would `(newVal !== oldVal)` make sense here?

File: src/modules/reducers/legacyBalances.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
+import store from "modules/store";
+import { fetchLegacyBalances, convertLegacyBalanceTx } from "modules/ethereum/legacyBalancesTransactions";
+
+export const LEGACY_BALANCE_REFRESH_REQUESTED = "legacyBalances/LEGACY_BALANCE_REFRESH_REQUESTED";
```
Comment (reyrodrigez) [inline src/modules/reducers/legacyBalances.js]: could you review which of these constants have to be actually exported?

File: src/modules/reducers/legacyBalances.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+export function dismissLegacyBalance(legacyTokenAddress) {
+    return async dispatch => {
+        dispatch({
+            type: LEGACY_BALANCE_DISMISS_REQUESTED,
+            request: { legacyTokenAddress }
+        });
+        try {
+            const contractBalances = [...store.getState().legacyBalances.contractBalances];
+            // TODO: shall we use tokenAddress as key instead of using an array?
```
Comment (reyrodrigez) [inline src/modules/reducers/legacyBalances.js]: you mean like here? https://github.com/Augmint/augmint-web/pull/149/files#diff-fdea7c4c7ec1e3df4eaf07c4d9247127R21

File: src/modules/reducers/rootReducer.js [hunk 4]
```diff
@@ -13,6 +13,7 @@ import exchange from "modules/reducers/exchange";
 import orders from "modules/reducers/orders";
 import trades from "modules/reducers/trades";
 import subscriptions from "modules/reducers/subscriptions";
+import legacyBalances from "modules/reducers/legacyBalances";
```
Comment (reyrodrigez) [inline src/modules/reducers/rootReducer.js]: why plural?

File: src/containers/app/LegacyTokens.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+                  );
+              })
+            : null;
+
+        return balances && balances.length > 0 && augmintToken.isConnected ? (
+            <Psegment>
+                <Container>
+                    <InfoPanel header="You have A-EUR in an older version of Augmint token contract">
+                        <p>
+                            There is newer Augmint A-EUR Token version deployed to the {network.name} network at{" "}
```
[the remark this answers is not in the document]
Comment (szerintedmi) [reply src/containers/app/LegacyTokens.js]: adds a space. this is how prettier autoformats when line breaks

File: src/modules/ethereum/legacyBalancesTransactions.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
+import store from "modules/store";
+import BigNumber from "bignumber.js";
+import { cost } from "./gas";
+import { processTx } from "modules/ethereum/ethHelper";
+//import { ONE_ETH_IN_WEI } from "../../utils/constants";
+//import AugmintToken from "contractsBuild/AugmintToken.json";
+import AugmintToken from "contractsBuild/TokenAEur.json";
```
[the remark this answers is not in the document]
Comment (szerintedmi) [reply src/modules/ethereum/legacyBalancesTransactions.js]: well, the idea that AugmintToken is the generic contract. TokenAEur is the first instance but there will be TokenAUsd etc. 
Might not be the best approach but it's how done everywhere else. I wouldn't jump to refactor everywhere rather do it when an other ccy implemented.

File: src/modules/legacyBalancesProvider.js (hunk 1 above)
Comment (szerintedmi) [reply src/modules/legacyBalancesProvider.js]: while loading it has ? value (see reducers/webConnect.js)

Comment (reyrodrigez) [reply src/modules/legacyBalancesProvider.js]: yeah, but there's no case when it changes after?

Comment (szerintedmi) [reply src/modules/legacyBalancesProvider.js]: TBH I don't remember why is it like that. maybe until intialised it's undefined or null then after initialisation it set to "?" which triggers change event?
 We should revisit all providers , it became convoluted and newer metamask / web3 versions might have made some parts unecessery

**Pull request merged**